PIL/WebPImagePlugin.py

```diff
@@ -30,17 +84,109 @@ class WebPImageFile(ImageFile.ImageFile):
+        size = None
+        lossy = True
+
+        first_chunk = True
+
+        while True:
+
+            chunk_header = self.fp.read(8)
+            if len(chunk_header) < 8:
+                # EOF.
+                break
+
+            chunk_fourcc = chunk_header[0:4]
+            chunk_size = i32le(chunk_header[4:8])
+
+            if first_chunk:
+                if chunk_fourcc not in _VP8_MODES_BY_IDENTIFIER:
+                    raise SyntaxError("not a WebP file")
+
+            if b'VP8X' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8x(chunk_size, self.fp)
+
+            elif b'VP8L' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8l(chunk_size, self.fp)
+                lossy = False
+
+            elif b'VP8 ' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8(chunk_size, self.fp)
+                lossy = True
+
+            elif b'ALPH' == chunk_fourcc:
+                mode = 'RGBA'
+
+            elif chunk_fourcc in _INFO_CHUNKS:
+                data = self.fp.read(chunk_size)
+                if len(data) < chunk_size:
+                    if ImageFile.LOAD_TRUNCATED_IMAGES:
+                        # Simulate EOF.
+                        break
+                    msg = "image file is truncated: incomplete %s chunk" % chunk_fourcc
+                    raise IOError(msg)
+                self.info[_INFO_CHUNKS[chunk_fourcc]] = data
+                chunk_size = 0
+
+            # Skip to next chunk.
+            self.fp.seek(chunk_size, os.SEEK_CUR)
+
+            first_chunk = False
+
+        if None in (mode, size, lossy):
+            raise SyntaxError("not a WebP file")
+
+        self.mode = mode
+        self.size = size
+        self.info['compression'] = 'lossy' if lossy else 'lossless'
+        self.tile = [('webp', (0, 0) + size, 0,
+                      # Decoder params: rawmode, has_alpha, width, height.
+                      (mode, 1 if 'RGBA' == mode else 0, size[0], size[1]))]
+
+        if not lossy:
+            # Incremental decoding on lossless is *really* slow, disable it.
+            self.decodermaxblock = 12 + i32le(header[4:8])
+            self.decoderconfig = (1,)
+
+    def load_prepare(self):
+        if not self.im:
+            # Hackety hack hack hack... force block storage (one continuous
+            # buffer), so we don't need to allocate a temporary output buffer
+            # for the decoder.
+            im = Image.core.new(self.mode, (0,0))
+            self.im = im.new_block(self.mode, self.size)
+        ImageFile.ImageFile.load_prepare(self)
+
+    def draft(self, mode, size):
+
+        if 1 != len(self.tile):
+            return
+
+        d, e, o, a = self.tile[0]
+
+        if mode in _VALID_WEBP_MODES:
+            a = mode, a[1], a[2], a[3]
+            self.mode = mode
+
+        if size:
```

Comment:
From the documentation:

> Configures the image file loader so it returns a version of the image that as closely as possible matches the given mode and size. For example, you can use this method to convert a color JPEG to greyscale while loading it, or to extract a 128x192 version from a PCD file

So if it's supported by the decoder, it should be allowed. I think converting – at the decoding stage – an RGB image to RGBA is precisely one of the use case of draft.

In fact one could even extend draft to support the WebP decoder ability to also do cropping on the fly (see the [Advanced Decoding API](https://developers.google.com/speed/webp/docs/api#advanced_decoding_api)).